paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/source/ContinuousFileSplitEnumerator.java

```diff
@@ -196,28 +198,39 @@ private void assignSplits() {
 
     private Map<Integer, List<FileStoreSourceSplit>> createAssignment() {
         Map<Integer, List<FileStoreSourceSplit>> assignment = new HashMap<>();
-        bucketSplits.forEach(
-                (bucket, splits) -> {
-                    if (splits.size() > 0) {
-                        // To ensure the order of consumption, the data of the same bucket is given
-                        // to a task to be consumed.
-                        int task = bucket % context.currentParallelism();
-                        if (readersAwaitingSplit.contains(task)) {
-                            // if the reader that requested another split has failed in the
-                            // meantime, remove
-                            // it from the list of waiting readers
-                            if (!context.registeredReaders().containsKey(task)) {
-                                readersAwaitingSplit.remove(task);
-                                return;
-                            }
-                            List<FileStoreSourceSplit> taskAssignment =
-                                    assignment.computeIfAbsent(task, i -> new ArrayList<>());
-                            if (taskAssignment.size() < splitBatchSize) {
-                                taskAssignment.add(splits.poll());
-                            }
-                        }
+        Deque<Integer> readersId = new ArrayDeque<>(readersAwaitingSplit);
+        for (Map.Entry<Integer, LinkedList<FileStoreSourceSplit>> entry : bucketSplits.entrySet()) {
+            Integer bucket = entry.getKey();
+            LinkedList<FileStoreSourceSplit> splits = entry.getValue();
+
+            if (splits.size() > 0 && readersId.size() > 0) {
```

Comment:
Here is bug: If the readers have come to wait, then at this time came a bunch of splits, the result here only one sent out. Other readers are still waiting.